File: DokanyAPI/DokanyApp.BLL/DTO/CartItemDTO.cs
﻿using System.Collections.Generic;

namespace DokanyApp.BLL
{
    public class CartItemDTO
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Quantity { get; set; }
        public decimal UnitCost { get; set; }
        public decimal Total { get; set; }
        public int ShoppingCartId { get; set; }
        public int ProductId { get; set; }
        public int CustomerId { get; set; }

        public User Customer { get; set; }
        public Product Product { get; set; }
        public ICollection<Order> Order { get; set; }
    }
}

File: Dokany UI/src/app/dtos/cartItemDto.ts
export class CartItemDto{
    Id: Number;
    Name: String;
    Quantity: Number;
    UnitCost: Number;
    Total: Number;
    ShoppingCartId: Number;
    ProductId: Number;
    CustomerId: Number;

    constructor(id: Number,
        name: String,
        quantity: Number,
        unitCost: Number,
        total: Number,
        shoppingCartId: Number,
        productId: Number,
        customerId: Number){
            this.Id = id;
            this.Name = name;
            this.Quantity = quantity;
            this.UnitCost = unitCost;
            this.Total = total;
            this.ShoppingCartId = shoppingCartId;
            this.ProductId = productId;
            this.CustomerId = customerId;
        }
}